pkg/driver/deployment_awslambda.go

```diff
@@ -22,46 +30,36 @@ func DeployFunctionsAWSLambda(functions []*common.Function) {
 		for parallelIndex := 0; parallelIndex < parallelDeployment; parallelIndex++ {
 			if i < len(functionGroups) {
 				wg.Add(1)
-				go func(functionGroup []*common.Function, index int) {
+				go func(functions []*common.Function, functionGroup []*common.Function, index int) {
 					defer wg.Done()
-					log.Debugf("Creating serverless-%d.yml", index)
-
-					// Create serverless.yml file
-					serverless := Serverless{}
-					serverless.CreateHeader(index, provider)
-					serverless.AddPackagePattern("!**")
-					serverless.AddPackagePattern("./server/trace-func-go/aws/trace_func")
-
-					for i := 0; i < len(functionGroup); i++ {
-						serverless.AddFunctionConfig(functionGroup[i], provider)
-					}
-
-					serverless.CreateServerlessConfigFile(index)
-
 					log.Debugf("Deploying serverless-%d.yml", index)
 					// Deploy serverless functions and update the function endpoints
 					functionToURLMapping := DeployServerless(index)
 
 					if functionToURLMapping == nil {
-						log.Fatalf("Failed to deploy serverless-%d.yml", index)
+						CleanAWSLambda(functions)                               // Clean up all deployed functions before exiting
```

Comment:
Go allows variable capture by reference in the lambdas. So, if it is the only reference to `functions` inside the goroutine, we can remove it from arguments, for clarity reasons.